pennylane/beta/tapes/qubit_param_shift.py

```diff
@@ -0,0 +1,220 @@
+
+    This results in :math:`4N + 1` evaluations.
+
+    In the case where :math:`O` is involutory (:math:`O^2 = I`), the first term in the above
+    expression vanishes, and we are simply left with
+
+    .. math:: \frac{\partial g}{\partial p_i} = - 2 f(p_i) \frac{\partial f}{\partial p_i},
+
+    allowing us to compute the gradient using :math:`2N + 1` evaluations.
+    """
+
+    def _update_circuit_info(self):
+        super()._update_circuit_info()
+
+        # set parameter_shift as the analytic_pd method
+        self.analytic_pd = self.parameter_shift
+
+        # check if the quantum tape contains any measurements
+        self.var_mask = [m.return_type is qml.operation.Variance for m in self.measurements]
+
+        if any(self.var_mask):
+            # The tape contains variances.
+            # Set parameter_shift_var as the analytic_pd method
+            self.analytic_pd = self.parameter_shift_var
+
+            # Make a copy of the original measurements; we will be mutating them
+            # during the parameter shift method.
+            self._original_measurements = self._measurements.copy()
+
+            # Finally, store the locations of any variance measurements in the
+            # measurement queue.
+            self.var_idx = np.where(self.var_mask)[0]
+
+    def _grad_method(self, idx, use_graph=True, default_method="F"):
+        op = self._par_info[idx]["op"]
+
+        if op.grad_method == "F":
+            return "F"
+
+        return super()._grad_method(idx, use_graph=use_graph, default_method="A")
+
+    def jacobian(self, device, params=None, **options):
+        # The parameter_shift_var method needs to evaluate the circuit
+        # at the unshifted parameter values; these are stored in the
+        # self._evA attribute. Here, we set the value of the attribute to None
+        # before each Jacobian call, so that the expectation value is calculated only once.
+        self._evA = None
+        return super().jacobian(device, params, **options)
+
+    def parameter_shift(self, idx, device, params, **options):
+        r"""Partial derivative using the parameter-shift rule of a tape consisting of *only*
+        expectation values of observables.
+
+        Args:
+            idx (int): trainable parameter index to differentiate with respect to
+            device (.Device, .QubitDevice): a PennyLane device
+                that can execute quantum operations and return measurement statistics
+            params (list[Any]): the quantum tape operation parameters
+
+        Keyword Args:
+            shift (float): the parameter shift value
+
+        Returns:
+            array[float]: 1-dimensional array of length determined by the tape output
+            measurement statistics
+        """
+        op = self._par_info[idx]["op"]
+        p_idx = self._par_info[idx]["p_idx"]
+
+        s = (
+            np.pi / 2
+            if op.grad_recipe is None or op.grad_recipe[p_idx] is None

```

Comment:
Is the second case for when a gate has multiple params?

Reply:
yep!